zegami_sdk/collection.py

```diff
@@ -367,6 +370,88 @@ def add_feature_pipeline(self, pipeline_name, steps, source=0, generate_snapshot
                 'source': source.name
             }
             self.add_snapshot(snapshot_name, snapshot_desc, snapshot_payload)
+    
+    def get_feature_pipelines(self):
+        """
+        Get all feature pipelines in a collection.
+
+        Example shape:
+            feature_pipelines = [
+                {
+                    name='mRMR20',
+                    source=0,
+                    steps=[     # list of nodes which would feed one into the other in sequence
+                        {
+                            'action': 'mRMR',
+                            'params': {
+                                'target_column': 'weight',
+                                'K': 20,
+                            },
+                        },
+                        {
+                            'action': 'cluster',
+                            'params': {
+                                "out_column_start_order": 1010,
+                                'algorithm_args': {
+                                    'algorithm': 'umap',
+                                    'n_components': 2,
+                                    "n_neighbors": 15,
+                                    "min_dist": 0.5,
+                                    "spread": 2,
+                                    "random_state": 123,
+                                }
+                            }
+                        }
+                    ]
+                },
+            ]
+        """
+        all_nodes = self.node_statuses
+        source_names = [s.name for s in self.sources]
+        feature_pipelines = []
+
+        # nodes sorted by source and feature pipeline name
+        feature_pipelines_nodes = {}
+        for source_name in source_names:
+            feature_pipelines_nodes[source_name] = {}
+
+        for node in all_nodes:
+            node_groups = node.get('node_groups')
+            # check if node_groups contain 'feature_pipeline_'
+            if node_groups and len(node_groups) == 3:
+                # get the source name after 'source_'
+                node_source_name = node_groups[0][7:]
+                # get the source name after 'feature_pipeline_'
+                feature_pipeline_name = node_groups[2][17:]
```

Comment:
Probably not safe to have these assumptions about the orders of the node groups.
Better to do something like
```
[group for group in node_groups if node_group.startswith('feature_pipeline')][0]
```